exp/runtime/internal/controllers/extensionconfig_controller.go

```diff
@@ -163,3 +202,41 @@ func discoverExtensionConfig(ctx context.Context, runtimeClient runtimeclient.Cl
 	conditions.MarkTrue(discoveredExtension, runtimev1.RuntimeExtensionDiscoveredCondition)
 	return discoveredExtension, nil
 }
+
+// reconcileCABundle reconciles the CA bundle for the ExtensionConfig.
+// cert-manager code: pkg/controller/cainjector/sources.go certificateDataSource (avoid cert-manager import via unstructured).
+func reconcileCABundle(ctx context.Context, client client.Client, config *runtimev1.ExtensionConfig) error {
+	secretNameRaw, ok := config.Annotations[caBundleInjectFromSecretAnnotation]
+	if !ok {
+		return nil
+	}
+	secretName := splitNamespacedName(secretNameRaw)
+
+	if secretName.Namespace == "" || secretName.Name == "" {
+		return errors.Errorf("secret name %q must be in the form namespace/name", secretNameRaw)
+	}
+
+	var secret corev1.Secret
+	// Note: this is an expensive API call because secrets are explicitly not cached.
+	if err := client.Get(ctx, secretName, &secret); err != nil {
+		return errors.Wrapf(err, "failed to get secret %s", secretNameRaw)
+	}
+
+	caData, hasCAData := secret.Data["ca.crt"]
```

Comment:
Should we put `ca.crt` in a local constant? (I wouldn't go so far and expose it as an API but a const is probably appropriate)